### PR TITLE
ISPN-4038 CustomFailurePolicyTxTest.testPrepareFailure random failures

### PR DESCRIPTION
**Pull request by pruivo**
https://issues.jboss.org/browse/ISPN-4038

### CODE REVIEW

File: core/src/test/java/org/infinispan/xsite/backupfailure/tx/BaseBackupTxFailureTest.java [hunk 1]
```diff
@@ -33,11 +32,10 @@ protected void createSites() {
       backup("LON").getAdvancedCache().getAsyncInterceptorChain().addInterceptor(failureInterceptor, 1);
    }
 
-   @Test(groups = {"xsite", "unstable"})
    public void testPrepareFailure() {
       failureInterceptor.enable();
       try {
-         Exceptions.expectException(XAException.class, () -> cache("LON", 0).put("k", "v"));
+         Exceptions.expectException(CacheException.class, () -> cache("LON", 0).put("k", "v"));
```
Comment (tristantarrant) [inline core/src/test/java/org/infinispan/xsite/backupfailure/tx/BaseBackupTxFailureTest.java]: You could use Exceptions.expectException(CacheException.class, RollbackException.class, () -> cache("LON", 0).put("k", "v"));
I noticed the XAException is now a suppressed exception inside the RollbackException, but exploring the whole exception hierarchy might be overkill

Comment (pruivo) [inline core/src/test/java/org/infinispan/xsite/backupfailure/tx/BaseBackupTxFailureTest.java]: The `XaException` are never seen by the user. they are only there for interaction between the `TransactionManager` and the `XaResource`/`Synchronization`. 
I'll change to check for `RollbackException`

**Comment by tristantarrant** [comment]
Good stuff. merged.

**Comment by tristantarrant** [comment]
Merged

**Pull request merged**